### DISCUSSION TITLE
Provide multiple vpn or wireguard connections for a single qBittorrent container

### DISCUSSION THREAD

**Discussion by sorryusernameisalreadytaken** [qdm12/gluetun#1455] [Q&A]
[It seems that qBittorrent is generally able to handle multiple ethernet devices.](https://github.com/qbittorrent/qBittorrent/issues/639#issuecomment-618426759) I need to make sure that my qBittorrent is not only provided one vpn / wireguard connection, but I need two.

So my goal is to have two network adapters available inside the qBittorrent container, qBittorrent can work with both of them and both network adapters run through my vpn provider each.

I am extremely unsure right now which approach to take for this. Docker networks can probably only provide networks between multiple containers.

Right now I'm using "network_mode: container:gluetun" to give the linuxserver/qbittorrent a connection only through the gluetun. I was wondering if I could map multiple tunnel devices.
E.g.:
gluetun1
/dev/net/tun:/dev/net/tun

gluetun2
/dev/net/tun:/dev/net/tun2

**Comment by mangotango69202** [comment]
bump
would also like this feature

**Comment by MrGangrene** [comment]
Also interested in this feature.
Different use case but similar enough.
I am running a bunch of services that use ridiculous bandwidth during peak office hours and our VPN provider is a bottleneck.
Was wondering if it is possible to set up multiple connections to PIA in a single gluetun container and have some of our containers round-robin the available connections.
Doesn't really matter if our data is sent to it's destination from different countries.

Ideally, I want gluetun to have a connection to fe, the Netherlands, Belgium, Luxembourg, France and Germany and have our containers routing through gluetun to use whatever connection is available.
This way we can route all that VPN traffic over a different VLAN that bypasses our UDM's DPI as this is also causing bottleneck issues because the office fibre is much faster than the DPI allows.

Thanks!

**Comment by ShayBox** [comment]
The best I've been able to do is to create 43 different identical containers and set up my own rotation service, it uses a lot of RAM though (~20GiB/Total). One container that can connect to and proxy multiple VPNs would be nice, I know it's possible but I'm not smart enough with Wire Guard and IP rules.

EDIT: Reducing the memory usage per container would be helpful though, currently they use way too much memory  (~400MiB/Each), maybe switch to Alpine?

EDIT2: I found this discussion https://github.com/qdm12/gluetun/issues/765
Adding the following reduced memory usage per container to around ~55MiB/Each ~6.3GiB/Total
- BLOCK_MALICIOUS=off
- BLOCK_SURVEILLANCE=off
- BLOCK_ADS=off
- DOT=off

**Reply by telgareith** [reply]
Can you provide some details? Preferably a compose file?
